File: app/src/main/java/com/example/tochkatest/MainActivity.java
package com.example.tochkatest;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Button;
import android.widget.ImageButton;

import androidx.appcompat.app.AppCompatActivity;

import com.facebook.AccessToken;
import com.facebook.CallbackManager;
import com.facebook.FacebookCallback;
import com.facebook.FacebookException;
import com.facebook.GraphRequest;
import com.facebook.GraphResponse;
import com.facebook.login.LoginResult;
import com.facebook.login.widget.LoginButton;
import com.google.android.gms.auth.api.signin.GoogleSignIn;
import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
import com.google.android.gms.auth.api.signin.GoogleSignInClient;
import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
import com.google.android.gms.common.api.ApiException;
import com.google.android.gms.tasks.Task;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Arrays;

public class MainActivity extends AppCompatActivity {

    //google
    final int RC_SIGN_IN = 0;
    private GoogleSignInClient mGoogleSignInClient;

    //fb
    private CallbackManager callbackManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //vk
        ImageButton vkImg = findViewById(R.id.vk);
        vkImg.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, StartActivity.class)));

        //google
        ImageButton googleImg = findViewById(R.id.gl);
        googleImg.setOnClickListener(v -> signIn());

        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
                .requestEmail()
                .build();
        mGoogleSignInClient = GoogleSignIn.getClient(this, gso);


        //fb
        LoginButton fbBtn = findViewById(R.id.fb);

        fbBtn.setReadPermissions(Arrays.asList( "public_profile"));
        // Creating CallbackManager
        callbackManager = CallbackManager.Factory.create();
        // Registering CallbackManager with the LoginButton
        fbBtn.registerCallback(callbackManager, new FacebookCallback<LoginResult>() {
//            AccessToken accessToken = AccessToken.getCurrentAccessToken();
//                if (accessToken != null) {
//                useLoginInformation(accessToken);
//            }
            @Override
            public void onSuccess(LoginResult loginResult) {
                // Retrieving access token using the LoginResult
                AccessToken accessToken = loginResult.getAccessToken();
                useLoginInformation(accessToken);

            }
            @Override
            public void onCancel() {
            }
            @Override
            public void onError(FacebookException error) {
            }
        });




        //exit
        Button exit = findViewById(R.id.exitBtn);
        exit.setOnClickListener(v -> System.exit(0));

    }


    //google
    private void signIn() {
        Intent signInIntent = mGoogleSignInClient.getSignInIntent();
        startActivityForResult(signInIntent, RC_SIGN_IN);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        // Result returned from launching the Intent from GoogleSignInClient.getSignInIntent(...);
        if (requestCode == RC_SIGN_IN) {
            // The Task returned from this call is always completed, no need to attach
            // a listener.
            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
            handleSignInResult(task);
        } else{
            //fb
            callbackManager.onActivityResult(requestCode, resultCode, data);
        }
    }

    //google
    private void handleSignInResult(Task<GoogleSignInAccount> completedTask) {
        try {
            GoogleSignInAccount account = completedTask.getResult(ApiException.class);

            // Signed in successfully, show authenticated UI.
            startActivity(new Intent(MainActivity.this, StartGoogleActivity.class));


        } catch (ApiException e) {
            // The ApiException status code indicates the detailed failure reason.
            // Please refer to the GoogleSignInStatusCodes class reference for more information.
            Log.e("TAG", "signInResult:failed code=" + e.getStatusCode());
        }
    }

    //fb
    private void useLoginInformation(AccessToken accessToken) {
        /**
         Creating the GraphRequest to fetch user details
         1st Param - AccessToken
         2nd Param - Callback (which will be invoked once the request is successful)
         **/
        GraphRequest request = GraphRequest.newMeRequest(accessToken, new GraphRequest.GraphJSONObjectCallback() {
            //OnCompleted is invoked once the GraphRequest is successful
            @Override
            public void onCompleted(JSONObject object, GraphResponse response) {
                try {


                    String nameFB = object.getString("name");
                    String imageFB = object.getJSONObject("picture").getJSONObject("data").getString("url");

                    Intent intent = new Intent(MainActivity.this, StartFBActivity.class);
                    intent.putExtra("nameFB",nameFB);
                    intent.putExtra("imageFB",imageFB);
                    startActivity(intent);


                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        });
        // We set parameters to the GraphRequest using a Bundle.
        Bundle parameters = new Bundle();
        parameters.putString("fields", "id,name,picture.width(200)");
        request.setParameters(parameters);
        // Initiate the GraphRequest
        request.executeAsync();
    }




}

File: app/src/main/java/com/example/tochkatest/view/MainActivity.kt
package com.example.tochkatest.view

import android.content.Intent
import android.os.Bundle
import android.widget.Button

import androidx.appcompat.app.AppCompatActivity
import com.example.tochkatest.R
import com.example.tochkatest.useCases.FacebookUseCase
import com.example.tochkatest.useCases.GoogleUseCase
import com.example.tochkatest.useCases.VkUseCase

import com.facebook.FacebookCallback
import com.facebook.FacebookException
import com.facebook.login.LoginResult
import com.facebook.login.widget.LoginButton
import com.google.android.gms.common.SignInButton

import java.util.Arrays

class MainActivity : AppCompatActivity() {


    private var googleUseCase: GoogleUseCase? = null
    private var facebookUseCase: FacebookUseCase? = null
    private var vkUseCase: VkUseCase? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        googleUseCase = GoogleUseCase(this)
        facebookUseCase = FacebookUseCase(this)
        vkUseCase = VkUseCase(this)

        //vk
        val vkImg = findViewById<Button>(R.id.vk)
        vkImg.setOnClickListener { vkUseCase!!.checkAuth() }

        //google
        val googleImg = findViewById<SignInButton>(R.id.gl)
        googleImg.setOnClickListener { googleUseCase!!.signIn(this) }

        //fb
        val fbBtn = findViewById<LoginButton>(R.id.fb)
        fbBtn.setReadPermissions(Arrays.asList("public_profile"))
        fbBtn.registerCallback(facebookUseCase!!.callbackManager, object : FacebookCallback<LoginResult> {
            override fun onSuccess(loginResult: LoginResult) {
                facebookUseCase!!.useLoginInformation(loginResult)
            }

            override fun onCancel() {}
            override fun onError(error: FacebookException) {}
        })

    }


    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (data != null) {
            googleUseCase!!.onActivityResult(requestCode, resultCode, data)
            vkUseCase!!.onActivityResult(requestCode, resultCode, data)
            facebookUseCase!!.onActivityResult(requestCode, resultCode, data)
        }



    }


}
